apps/condo/domains/resident/schema/ServiceConsumer.js

```diff
@@ -122,6 +123,30 @@ const ServiceConsumer = new GQLListSchema('ServiceConsumer', {
         delete: false,
         auth: true,
     },
+    hooks: {
+        afterChange: async ({ context, operation, updatedItem, originalInput }) => {
+            const deletedAt = get(updatedItem, 'deletedAt', null)
+            const sender = get(originalInput, 'sender', null)
+            const dv = get(originalInput, 'dv', null)
+
+            // handle soft delete
+            // in order to soft delete recurrent payment contexts
+            if (operation === 'update' && deletedAt) {
```

Comment:
may be you need to extract some serviceSchema helter for that. Look at Ticket hook handlers examples

Reply:
I have extracted function for delete orphans recurrentContexts into acquiring/utils/serverSchema/helpers.js